tests/api/test_shop.py

```diff
@@ -830,3 +830,154 @@ def test_mutation_update_company_address_remove_address_without_address(
+    )
+    content = get_graphql_content(response)
+
+    assert content["data"]["shop"]["staffNotificationRecipient"] == [
+        {
+            "active": True,
+            "email": staff_user.email,
+            "user": {
+                "firstName": staff_user.first_name,
+                "lastName": staff_user.last_name,
+                "email": staff_user.email,
+            },
+        }
+    ]
+
+
+def test_staff_notification_create_mutation(
+    staff_api_client, staff_user, permission_manage_settings
+):
+    query = """
+        mutation StaffNotificationRecipient ($user: ID!) {
+            staffNotificationRecipientCreate(input: {user: $user}) {
+                staffNotificationRecipient {
+                    active
+                    email
+                    user {
+                        firstName
+                        lastName
+                        email
+                    }
+                }
+            }
+        }
+    """
+    user_id = graphene.Node.to_global_id("User", staff_user.id)
+    variables = {"user": user_id}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_settings]
+    )
+    content = get_graphql_content(response)
+
+    assert content["data"]["staffNotificationRecipientCreate"] == {
+        "staffNotificationRecipient": {
+            "active": True,
+            "email": staff_user.email,
+            "user": {
+                "firstName": staff_user.first_name,
+                "lastName": staff_user.last_name,
+                "email": staff_user.email,
+            },
+        }
+    }
+
+
+def test_staff_notification_create_mutation_with_staffs_email(
+    staff_api_client, staff_user, permission_manage_settings
+):
+    query = """
+        mutation StaffNotificationRecipient ($staff_email: String!) {
+            staffNotificationRecipientCreate(input: {staffEmail: $staff_email}) {
+                staffNotificationRecipient {
+                    active
+                    email
+                    user {
+                        id
+                        firstName
+                        lastName
+                        email
+                    }
+                }
+            }
+        }
+    """
+    user_id = graphene.Node.to_global_id("User", staff_user.id)
+    variables = {"staff_email": staff_user.email}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_settings]
+    )
+    content = get_graphql_content(response)
+
+    assert content["data"]["staffNotificationRecipientCreate"] == {
+        "staffNotificationRecipient": {
+            "active": True,
+            "email": staff_user.email,
+            "user": {
+                "id": user_id,
+                "firstName": staff_user.first_name,
+                "lastName": staff_user.last_name,
+                "email": staff_user.email,
+            },
```

Comment:
It would be nice to test the case where there is no staff having such email. `user` will be `None`. Or we should remove the association all together when no user is selected.

Reply:
I agree that this case should be tested as well.

Reply:
done